javascript-loops/main.js

```diff
@@ -0,0 +1,86 @@
+/* exported getNumbersToTen,
+            getEvenNumbersToTwenty,
+            repeatWord,
+            logEachCharacter,
+            doubleAll,
+            getKeys,
+            getValues
+ */
+
+function getNumbersToTen() {
+  var i = 1;
+  var numbers = [];
+  var currentNumber = 1;
+  while (i < 11) {
+    numbers.push(currentNumber);
+    currentNumber += 1;
+    i++;
+  }
+  return numbers;
+}
+console.log('numbers 1 to 10', getNumbersToTen());
+
+function getEvenNumbersToTwenty() {
+  var i = 2;
+  var numbers = [];
+  var currentNumber = 2;
+  while (i < 21) {
+    numbers.push(currentNumber);
+    currentNumber += 2;
+    i = i + 2;
+  }
+  return numbers;
+}
+console.log('Even numbers 2 to 20', getEvenNumbersToTwenty());
+
+function repeatWord(word, times) {
+  var count = 1;
+  var repeated = '';
+  while (count < times + 1) {
+    repeated += ' ' + word;
+    count += 1;
+  }
+  return repeated;
+}
+console.log('Repeat function:', repeatWord('JAVASCRIPTTT', 22));
+
+function logEachCharacter(string) {
+  var final = '';
+  for (var i = 0; i < string.length; i++) {
+    (final = string[i]);
+    console.log('Log each char of string:', final);
+  }
+}
+logEachCharacter('hDSFN');
+
+function doubleAll(numbers) {
+  var doubled = [];
+  for (var i = 0; i < numbers.length; i++) {
+    numbers[i] = numbers[i] * 2;
```

Comment:
If you receive a reference data type as an argument for a function, you should NEVER change the values in the original, unless that is the intended purpose of the function. By updating the value in the original array, now the array given as an argument to this function would have its values permanently doubled, and that could cause issues if the array still needs to be used again elsewhere (and data is often used in multiple places in your code)